src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java

```diff
@@ -72,16 +73,20 @@ public void setBindingManager(ProjectBindingManager bindingManager) {
   }
 
   public void initialize(@Nullable List<WorkspaceFolder> workspaceFolders) {
+//    this.workspaceFolders = workspaceFolders;
     if (workspaceFolders != null) {
       workspaceFolders.forEach(wf -> {
         var uri = create(wf.getUri());
         addFolder(wf, uri);
       });
-      executor.submit(new CatchingRunnable(() -> backendServiceFacade.getBackendService().addWorkspaceFolders(workspaceFolders, getBindingProvider()),
-        t -> logOutput.errorWithStackTrace("Failed to initialize workspace folders.", t)));
     }
   }
 
+  public void initialized() {
+    executor.submit(new CatchingRunnable(() -> backendServiceFacade.getBackendService().addWorkspaceFolders(this.workspaceFolders, getBindingProvider()),
+      t -> logOutput.errorWithStackTrace("Failed to initialize workspace folders.", t)));
+  }
+
```

Comment:
Why moving this? I think it should work how it was before, since `WorkspaceFoldersManager#initialize` is called after `BackendServiceFacade.initialize`

Reply:
This needs all settings to be initialized first, which happens during `initialized` phase (after backend initialization)